.github/workflows/build-wheels.yml

```diff
@@ -0,0 +1,146 @@
+name: Build and test wheels
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - release-*
+      - "*wheel*" # must quote since "*" is a YAML reserved character; we want a string
+    tags:
+      - "*"
+  pull_request:
+    branches:
+      - "*wheel*" # must quote since "*" is a YAML reserved character; we want a string
+
+jobs:
+  build_wheels:
+    name: Wheel ${{ matrix.buildplat[0] }}-${{ matrix.buildplat[1] }}-${{ matrix.python }}
+    runs-on: ${{ matrix.buildplat[0] }}
+    strategy:
+      matrix:
+        buildplat:
+          - [ubuntu-22.04, manylinux_x86_64]
+          - [macos-13, macosx_x86_64]
+          - [macos-14, macosx_arm64]
+          - [windows-2022, win_amd64]
+        python: ["cp39", "cp310", "cp311", "cp312"]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: "Brew setup on macOS" # x-ref c8e49ba8f8b9ce
+        if: ${{ startsWith(matrix.os, 'macos-') == true }}
+        run: |
+          set -e pipefail
+          brew update
+          brew install automake pkg-config ninja llvm
+
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.18.1
+        env:
+          CIBW_BUILD_VERBOSITY: 3
+          CIBW_ENVIRONMENT_MACOS: >
+            CC=clang
+            CXX=clang++
+          MACOSX_DEPLOYMENT_TARGET: "11.0"
+          CIBW_ARCHS: all
+          CIBW_PRERELEASE_PYTHONS: True
+          CIBW_BUILD: ${{ matrix.python }}-${{ matrix.buildplat[1] }}
+          # __init__.py interferes with the tests and is included as local file instead of
+          # used from wheels. To be honest, tests should not be in the source folder at all.
+          CIBW_BEFORE_TEST: rm {project}/tiledb/__init__.py
+        with:
+          output-dir: wheelhouse
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: cibw-wheels-${{ matrix.buildplat[1] }}-${{ strategy.job-index }}
+          path: "./wheelhouse/*.whl"
+
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-latest
+    outputs:
+      sdist_name: ${{ steps.get_sdist_name.outputs.sdist_name }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Build sdist
+        run: pipx run build --sdist
+
+      - name: Get sdist package name
+        id: get_sdist_name
+        run: |
+          echo "sdist_name=$(ls dist/ | head -n 1)" >> "$GITHUB_OUTPUT"
+
+      - uses: actions/upload-artifact@v4
+        with:
+          name: sdist
+          path: dist/*.tar.gz
+
+  test_sdist:
```

Comment:
why are we testing sdist but not wheels?

Reply:
We are also testing wheels, but its automatically done by cibuildwheel step